google-cloud-storage/src/main/java/com/google/cloud/storage/BlobWriteChannel.java

```diff
@@ -83,14 +84,52 @@ private StorageObject getRemoteStorageObject() {
         .get(getEntity().toPb(), Maps.newEnumMap(StorageRpc.Option.class));
   }
 
-  private StorageException unrecoverableState(
-      int chunkOffset, int chunkLength, long localPosition, long remotePosition, boolean last) {
+  private static StorageException unrecoverableState(
+      String uploadId,
+      int chunkOffset,
+      int chunkLength,
+      long localPosition,
+      long remotePosition,
+      boolean last) {
+    return unrecoverableState(
+        uploadId,
+        chunkOffset,
+        chunkLength,
+        localPosition,
+        remotePosition,
+        last,
+        "Unable to recover in upload.\nThis may be a symptom of multiple clients uploading to the same upload session.");
```

Comment:
I don't think multiple clients uploading to the same session is possible? The session should be created by the client and then only used by that client AFAIK.

Reply:
@frankyn Do you have context on why this was the original message? (I refactored it to a parameter so I could reuse the details formatting code).

Reply:
It's possible if the upload URI is shared between clients. GCS API supports it, however it's very unlikely it's a possible case.